packages/database/prisma/schema.prisma

```diff
@@ -28,74 +28,107 @@ enum BookingStatus {
 
 model Location {
   id           Int    @id @default(autoincrement())
-  locationName String
-  guideId      Int
+  locationName String @unique
 
-  guide Guide @relation(fields: [guideId], references: [id])
+  GuideLocation GuideLocation[]
+  PostLocation  PostLocation[]
 }
 
 model TourStyle {
   id            Int    @id @default(autoincrement())
-  tourStyleName String
-  guideId       Int
+  tourStyleName String @unique
+
+  GuideTourStyle GuideTourStyle[]
+}
+
+model Guide {
+  id             Int              @id @default(autoincrement())
+  userId         Int              @unique
+  certificateId  String
+  paymentId      String
+  user           User             @relation(fields: [userId], references: [id], onUpdate: Cascade, onDelete: Cascade)
+  reviews        Review[]
+  Bookings       Booking[]
+  GuideLocation  GuideLocation[]
+  GuideTourStyle GuideTourStyle[]
+}
+
+model GuideLocation {
+  guideId    Int
+  locationId Int
+
+  guide    Guide    @relation(fields: [guideId], references: [id], onUpdate: Cascade, onDelete: Cascade)
+  location Location @relation(fields: [locationId], references: [id], onUpdate: Cascade, onDelete: Cascade)
 
-  guide Guide @relation(fields: [guideId], references: [id])
+  @@id([guideId, locationId])
+}
+
+model GuideTourStyle {
+  guideId     Int
+  tourStyleId Int
+
+  guide     Guide     @relation(fields: [guideId], references: [id], onUpdate: Cascade, onDelete: Cascade)
+  tourStyle TourStyle @relation(fields: [tourStyleId], references: [id], onUpdate: Cascade, onDelete: Cascade)
+
+  @@id([guideId, tourStyleId])
 }
 
 model Review {
   id          Int      @id @default(autoincrement())
-  publishDate DateTime
+  publishDate DateTime @default(now())
   score       Decimal  @db.Decimal(2, 1)
-  text        String?
+  text        String
   reviewerId  Int
   guideId     Int
 
-  reviewer User  @relation(fields: [reviewerId], references: [id])
-  guide    Guide @relation(fields: [guideId], references: [id])
+  reviewer User  @relation(fields: [reviewerId], references: [id], onUpdate: Cascade, onDelete: Cascade)
+  guide    Guide @relation(fields: [guideId], references: [id], onUpdate: Cascade, onDelete: Cascade)
 }
 
 model Post {
-  id        Int      @id @default(autoincrement())
-  createdAt DateTime @default(now())
-  updatedAt DateTime @updatedAt
-  title     String
-  content   String?
-  authorId  Int
-  tags      String[]
-  fee       Decimal  @db.Decimal(10, 2)
-
-  author  User      @relation(fields: [authorId], references: [id])
-  bookings Booking[]
+  id             Int      @id @default(autoincrement())
+  createdAt      DateTime @default(now())
+  updatedAt      DateTime @updatedAt
+  title          String
+  content        String
+  authorId       Int
+  tags           String[]
+  fee            Decimal  @db.Decimal(10, 2)
+  maxParticipant Int
+  contactInfo    String
+
+  author       User           @relation(fields: [authorId], references: [id], onUpdate: Cascade, onDelete: Cascade)
```

Comment:
should an author be a guide instead

Reply:
we may refactor this in the future, as now frontend want our api : )